x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/calculations/utils.ts

```diff
@@ -41,6 +43,76 @@ export function checkForDateHistogram(layer: IndexPatternLayer, name: string) {
   ];
 }
 
+export function checkReferences(layer: IndexPatternLayer, columnId: string) {
+  const column = layer.columns[columnId] as ReferenceBasedIndexPatternColumn;
+
+  const errors: string[] = [];
+
+  column.references.forEach((referenceId, index) => {
+    if (!layer.columns[referenceId]) {
+      errors.push(
+        i18n.translate('xpack.lens.indexPattern.missingReferenceError', {
+          defaultMessage: '"{dimensionLabel}" is not fully configured',
+          values: {
+            dimensionLabel: column.label,
+          },
+        })
+      );
+    } else {
+      const referenceColumn = layer.columns[referenceId]!;
+      const requirements =
+        // @ts-expect-error not statically analyzed
+        operationDefinitionMap[column.operationType].requiredReferences[index];
```

Comment:
`@ts-expect-error` can make refactorings harder in the future, what about this approach?
```ts
      const definition = operationDefinitionMap[column.operationType];
      if (definition.input !== 'fullReference') {
        throw new Error('inconsistent state - column is not a reference operation');
      }
      const requirements = definition.requiredReferences[index];
```